src/components/modules/checkbox/checkbox.jsx

```diff
@@ -23,57 +19,44 @@ export default class Checkbox extends Component {
     };
 
     static defaultProps = {
-        component: 'div',
-        defaultClasses: true
+        ...DefaultProps.defaultProps
     };
 
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            active: this.props.checked
-        }
-    }
+    onClick = (event) => {
+        if (this.props.disabled || this.props.readOnly) return;
 
-    onClick() {
-        if (!this.state.active || (this.state.active && !this.props.radio)) {
-            this.setState({
-                active: !this.state.active
-            });
-        }
-    }
+        this.props.onClick(event);
```

Comment:
need to have stub handler in defaultProps: 
```
defaultProps = {
  ...DefaultProps.defaultProps,
  onClick: () => { }
}
```
otherwise it will fail if not provided or forgot